### PR TITLE
Pass crashReporter instance to RustLog

### PR DESCRIPTION
**Pull request by grigoryk**
Requires https://github.com/mozilla-mobile/android-components/pull/7257

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 1]
```diff
@@ -274,7 +274,7 @@ open class FenixApplication : LocaleAwareApplication() {
         return GlobalScope.async(Dispatchers.IO) {
             // ... but RustHttpConfig.setClient() and RustLog.enable() can be called later.
             RustHttpConfig.setClient(lazy { components.core.client })
-            RustLog.enable()
+            RustLog.enable(components.analytics.crashReporter)
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:277]: @grigoryk I just approved the AC PR and asked bors to land it. But can you confirm that those log messages are unlikely to contain PII?

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/FenixApplication.kt:277]: Error ones are definitely quite unlikely to contain PII. We already send them to Sentry in Firefox iOS, too. I'll double check with the a-s team, but I think we should be okay here.

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/FenixApplication.kt:277]: @eoger we're about to enable sending `error` log messages to Sentry. You don't expect them to have any PII, right?

Comment (eoger) [reply app/src/main/java/org/mozilla/fenix/FenixApplication.kt:277]: Yes [this should](https://github.com/mozilla/application-services/issues/2407) [be fine](https://github.com/mozilla/application-services/issues/909).

**Pull request merged**